scripts/move_session_workspace.py

```
@@ -0,0 +1,667 @@
+async def _can_defer_constraints(session: AsyncSession) -> bool:
+    """Return True if the current role can ALTER the ``sessions`` table's constraints.
+
+    Heuristic: if the probe is wrong, the apply transaction rolls back (pg_dump-backed),
+    so a false positive is safe — the ALTER will fail and the outer transaction aborts.
+    """
+    result = await session.scalar(
+        text(
+            "SELECT COALESCE("
+            + "  (SELECT rolsuper FROM pg_roles WHERE rolname = current_user)"
+            + "  OR pg_catalog.pg_get_userbyid(relowner) = current_user, false)"
+            + " FROM pg_class WHERE relname = 'sessions' AND relkind = 'r'"
```

Ruff (0.15.20)

[error] Possible SQL injection vector through string-based query construction

(S608)